core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -211,6 +211,11 @@ public long getSizeInBytes()
                 (rleValue != null ? rleValue.getSizeInBytes() : 0);
     }
 
+    public long getDirectSizeInBytes()
```

Comment:
this should most likely return `OptionalLong` and only for Rle

Reply:
I think we still want to count the current size in bytes for direct appenders towards the total direct size, but with a special case for RLE where we want to report the size "as if it were flattened". Dictionaries of course are still unavoidably under-reported here.